core/host.go

```diff
@@ -24,6 +25,33 @@ func (*Host) TypeDescription() string {
 	return "Information about the host environment."
 }
 
+// Lookup an environment variable in the host system from the current context
+func (Host) GetEnv(ctx context.Context, name string) string {
+	dag, err := CurrentDagqlServer(ctx)
+	if err != nil {
+		return ""
+	}
+	var plaintext string
+	err = dag.Select(ctx, dag.Root(), &plaintext,
+		dagql.Selector{
+			Field: "secret",
```

Comment:
lol sneaky

This works obviously but it might be wise to cut out the DagQL overhead and cut straight to the Buildkit plumbing:

https://github.com/dagger/dagger/blob/7c41c975fc9ee293510f3871fd9cfb8ae3c45a4e/core/secret.go#L164-L187

Reply:
You're right that the reference to "secret" is a little confusing. Ideally we would have a distinct session attachable just for reading env variables :) But then would have to make sure they don't leak in the traces...